File: cp_day2/linearSearch.js
const linearSearch = (find, arrayToSearch) => {
  if (find.toString().length === 1) {
    return charSearch(find, arrayToSearch)
  }
}

function charSearch(char, array) {
  for (let i = 0; i < array.length; i++) {
    if (array[i] === char) {
      return i
    }
  }
}

console.log(linearSearch(1, [5,3,4,1])) // return 3
console.log(linearSearch('s', ['a','s','f'])) // return 1

File: week_1/cp_day3/linear_search.py
# given a character or int, check if int/char in provided array
def linear_search(find, list):
    for index in range(len(list)):
        if list[index] == find:
            return index            

def global_linear_search(find, list):
    indices = []
    for index in range(len(list)):
        if find == list[index]:
            indices.append(index)
    return indices


# print(linear_search(4, [1,2,3,4]))
# print(linear_search('a', ['ad','s','d','f','g','h','j','a']))
print(global_linear_search(4, [1,2,3,4,4,1,2,4]))
print(global_linear_search('a', ['a','s','d','f','g','h','j','a']))